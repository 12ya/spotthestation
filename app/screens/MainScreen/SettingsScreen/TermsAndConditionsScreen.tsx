/* eslint-disable react-native/no-inline-styles */
/* eslint-disable @typescript-eslint/no-unsafe-argument */
/* eslint-disable @typescript-eslint/no-unsafe-member-access */
import { useNavigation } from "@react-navigation/native"
import { observer } from "mobx-react-lite"
import React, { useMemo } from "react"
import { ViewStyle, TextStyle, ScrollView, Pressable, View, Platform } from "react-native"
import { useSafeAreaInsets } from "react-native-safe-area-context"
import { Icon, Screen, Text } from "../../../components"
import { colors, fontSizes, lineHeights, scale, typography } from "../../../theme"
import en from "../../../i18n/en"
import { TxKeyPath, translate } from "../../../i18n"
import DeviceInfo from "react-native-device-info"

export const TermsAndConditionsScreen = observer(function TermsAndConditionsScreen() {
  const navigation = useNavigation()
  const topInset = useSafeAreaInsets().top

  const $headerStyleOverride: TextStyle = {
    top: topInset + scale(24),
  }

  const tnc = useMemo(() => {
    const platform = en.settings.termsAndConditionsData[Platform.OS]
    const translationPrefix = `settings.termsAndConditionsData.${Platform.OS}`
    const body: string[] = Object.keys(platform.body)

    return {
      title: translate(`${translationPrefix}.title` as TxKeyPath),
      intro1: translate(`${translationPrefix}.intro1` as TxKeyPath),
      intro2: translate(`${translationPrefix}.intro2` as TxKeyPath),
      intro3: translate(`${translationPrefix}.intro3` as TxKeyPath),
      appData: [
        `${translate(
          `${translationPrefix}.appData.line1` as TxKeyPath,
        )} ${DeviceInfo.getApplicationName()}`,
        `${translate(
          `${translationPrefix}.appData.line2` as TxKeyPath,
        )} ${DeviceInfo.getVersion()}`,
        translate(`${translationPrefix}.appData.line3` as TxKeyPath),
      ],
      body: body.map((item) => translate(`${translationPrefix}.body.${item}` as TxKeyPath)),
      contactData:
        Platform.OS === "ios"
          ? Object.keys(platform.contactData).map((item) =>
              translate(`${translationPrefix}.contactData.${item}` as TxKeyPath),
            )
          : null,
    }
  }, [en])

  return (
    <Screen
      preset="fixed"
      contentContainerStyle={[$container, $headerStyleOverride]}
      style={{ backgroundColor: colors.palette.neutral900 }}
      statusBarStyle="light"
    >
      <Pressable
        accessible
        accessibilityLabel="Back button"
        accessibilityHint="Navigates to the previous screen"
        accessibilityRole="button"
        onPress={() => navigation.goBack()}
        style={$backButton}
      >
        <Icon icon="caretLeft" color={colors.palette.neutral250} size={24} />
        <Text tx="settings.termsAndConditionsData.backButton" style={$backButtonText} />
      </Pressable>
      <ScrollView
        accessible
        accessibilityLabel="terms And Conditions scrollable us area"
        accessibilityHint="terms And Conditions scrollable us area"
        accessibilityRole="scrollbar"
        style={$scrollContainer}
        scrollEnabled
        contentContainerStyle={$scrollSontentContainerStyle}
      >
        <Pressable>
          <View
            accessible
            accessibilityLabel="terms And Conditions"
            accessibilityHint="terms And Conditions"
            accessibilityRole="text"
          >
            <Text text={tnc.title} style={$title} />
            <Text text={tnc.intro1} style={$text} />
            {tnc.appData.map((item) => (
              <Text key={item} text={item} style={[$text, { paddingBottom: 0 }]} />
            ))}
            <Text text="" style={$text} />
            {tnc.contactData &&
              tnc.contactData.map((item) => (
                <Text key={item} text={item} style={[$text, { paddingBottom: 0 }]} />
              ))}
            <Text text={tnc.intro2} style={$text} />
            <Text text={tnc.intro3} style={$text} />
            {tnc.body.map((item) => (
              <Text key={item} text={item} style={$text} />
            ))}
          </View>
        </Pressable>
      </ScrollView>
    </Screen>
  )
})

const $container: ViewStyle = {
  flex: 1,
  backgroundColor: colors.backgroundDark,
  height: "100%",
}

const $backButton: ViewStyle = {
  flexDirection: "row",
  alignItems: "center",
  paddingBottom: 11,
  paddingLeft: 16,
}

const $scrollSontentContainerStyle: ViewStyle = {
  flexGrow: 1,
  paddingBottom: 60,
}

const $scrollContainer: ViewStyle = {
  paddingHorizontal: 18,
}

const $title: TextStyle = {
  fontFamily: typography.primary?.normal,
  fontSize: fontSizes[36],
  lineHeight: lineHeights[44],
  color: colors.palette.neutral250,
  textAlign: "left",
  paddingBottom: 24,
}

const $text: TextStyle = {
  fontFamily: typography.primary?.normal,
  fontSize: fontSizes[24],
  lineHeight: lineHeights[34],
  color: colors.palette.neutral250,
  textAlign: "left",
  paddingBottom: 16,
}

const $backButtonText: TextStyle = {
  ...$text,
  color: colors.palette.neutral250,
  paddingBottom: 0,
  paddingLeft: 5,
}
