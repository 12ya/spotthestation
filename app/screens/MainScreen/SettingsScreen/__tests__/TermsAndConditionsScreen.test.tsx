/* eslint-disable react/display-name */
/* eslint-disable @typescript-eslint/no-unsafe-call */
/* eslint-disable @typescript-eslint/no-unsafe-member-access */
import React from "react"
import renderer from 'react-test-renderer'
import { TermsAndConditionsScreen } from '../TermsAndConditionsScreen'
import { NavigationContainer } from "@react-navigation/native"

it('renders correctly', () => {
  const tree = renderer
    .create(<NavigationContainer><TermsAndConditionsScreen /></NavigationContainer>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
