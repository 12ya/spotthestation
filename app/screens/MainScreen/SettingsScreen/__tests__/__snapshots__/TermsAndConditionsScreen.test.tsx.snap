// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <View
    accessibilityHint="Navigates to the previous screen"
    accessibilityLabel="Back button"
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingBottom": 11,
        "paddingLeft": 16,
      }
    }
  >
    <div />
    <div
      style={
        Object {
          "color": "#DEDDDE",
          "fontFamily": "Helvetica Neue",
          "fontSize": 35,
          "lineHeight": 49,
          "paddingBottom": 0,
          "paddingLeft": 5,
          "textAlign": "left",
        }
      }
    />
  </View>
  <RCTScrollView
    accessibilityHint="terms And Conditions scrollable us area"
    accessibilityLabel="terms And Conditions scrollable us area"
    accessibilityRole="scrollbar"
    accessible={true}
    contentContainerStyle={
      Object {
        "flexGrow": 1,
        "paddingBottom": 60,
      }
    }
    scrollEnabled={true}
    style={
      Object {
        "paddingHorizontal": 18,
      }
    }
  >
    <View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          accessibilityHint="terms And Conditions"
          accessibilityLabel="terms And Conditions"
          accessibilityRole="text"
          accessible={true}
        >
          <div
            style={
              Object {
                "color": "#DEDDDE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 52,
                "lineHeight": 63,
                "paddingBottom": 24,
                "textAlign": "left",
              }
            }
          >
            settings.termsAndConditionsData.ios.title undefined
          </div>
          <div
            style={
              Object {
                "color": "#DEDDDE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 35,
                "lineHeight": 49,
                "paddingBottom": 16,
                "textAlign": "left",
              }
            }
          >
            settings.termsAndConditionsData.ios.intro1 undefined
          </div>
          <div
            style={
              Array [
                Object {
                  "color": "#DEDDDE",
                  "fontFamily": "Helvetica Neue",
                  "fontSize": 35,
                  "lineHeight": 49,
                  "paddingBottom": 16,
                  "textAlign": "left",
                },
                Object {
                  "paddingBottom": 0,
                },
              ]
            }
          >
            settings.termsAndConditionsData.ios.appData.line1 undefined Test
          </div>
          <div
            style={
              Array [
                Object {
                  "color": "#DEDDDE",
                  "fontFamily": "Helvetica Neue",
                  "fontSize": 35,
                  "lineHeight": 49,
                  "paddingBottom": 16,
                  "textAlign": "left",
                },
                Object {
                  "paddingBottom": 0,
                },
              ]
            }
          >
            settings.termsAndConditionsData.ios.appData.line2 undefined 0.0.1
          </div>
          <div
            style={
              Array [
                Object {
                  "color": "#DEDDDE",
                  "fontFamily": "Helvetica Neue",
                  "fontSize": 35,
                  "lineHeight": 49,
                  "paddingBottom": 16,
                  "textAlign": "left",
                },
                Object {
                  "paddingBottom": 0,
                },
              ]
            }
          >
            settings.termsAndConditionsData.ios.appData.line3 undefined
          </div>
          <div
            style={
              Object {
                "color": "#DEDDDE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 35,
                "lineHeight": 49,
                "paddingBottom": 16,
                "textAlign": "left",
              }
            }
          />
          <div
            style={
              Array [
                Object {
                  "color": "#DEDDDE",
                  "fontFamily": "Helvetica Neue",
                  "fontSize": 35,
                  "lineHeight": 49,
                  "paddingBottom": 16,
                  "textAlign": "left",
                },
                Object {
                  "paddingBottom": 0,
                },
              ]
            }
          >
            settings.termsAndConditionsData.ios.contactData.line1 undefined
          </div>
          <div
            style={
              Array [
                Object {
                  "color": "#DEDDDE",
                  "fontFamily": "Helvetica Neue",
                  "fontSize": 35,
                  "lineHeight": 49,
                  "paddingBottom": 16,
                  "textAlign": "left",
                },
                Object {
                  "paddingBottom": 0,
                },
              ]
            }
          >
            settings.termsAndConditionsData.ios.contactData.line2 undefined
          </div>
          <div
            style={
              Array [
                Object {
                  "color": "#DEDDDE",
                  "fontFamily": "Helvetica Neue",
                  "fontSize": 35,
                  "lineHeight": 49,
                  "paddingBottom": 16,
                  "textAlign": "left",
                },
                Object {
                  "paddingBottom": 0,
                },
              ]
            }
          >
            settings.termsAndConditionsData.ios.contactData.line3 undefined
          </div>
          <div
            style={
              Array [
                Object {
                  "color": "#DEDDDE",
                  "fontFamily": "Helvetica Neue",
                  "fontSize": 35,
                  "lineHeight": 49,
                  "paddingBottom": 16,
                  "textAlign": "left",
                },
                Object {
                  "paddingBottom": 0,
                },
              ]
            }
          >
            settings.termsAndConditionsData.ios.contactData.line4 undefined
          </div>
          <div
            style={
              Array [
                Object {
                  "color": "#DEDDDE",
                  "fontFamily": "Helvetica Neue",
                  "fontSize": 35,
                  "lineHeight": 49,
                  "paddingBottom": 16,
                  "textAlign": "left",
                },
                Object {
                  "paddingBottom": 0,
                },
              ]
            }
          >
            settings.termsAndConditionsData.ios.contactData.line5 undefined
          </div>
          <div
            style={
              Object {
                "color": "#DEDDDE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 35,
                "lineHeight": 49,
                "paddingBottom": 16,
                "textAlign": "left",
              }
            }
          >
            settings.termsAndConditionsData.ios.intro2 undefined
          </div>
          <div
            style={
              Object {
                "color": "#DEDDDE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 35,
                "lineHeight": 49,
                "paddingBottom": 16,
                "textAlign": "left",
              }
            }
          >
            settings.termsAndConditionsData.ios.intro3 undefined
          </div>
          <div
            style={
              Object {
                "color": "#DEDDDE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 35,
                "lineHeight": 49,
                "paddingBottom": 16,
                "textAlign": "left",
              }
            }
          >
            settings.termsAndConditionsData.ios.body.line1 undefined
          </div>
          <div
            style={
              Object {
                "color": "#DEDDDE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 35,
                "lineHeight": 49,
                "paddingBottom": 16,
                "textAlign": "left",
              }
            }
          >
            settings.termsAndConditionsData.ios.body.line2 undefined
          </div>
          <div
            style={
              Object {
                "color": "#DEDDDE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 35,
                "lineHeight": 49,
                "paddingBottom": 16,
                "textAlign": "left",
              }
            }
          >
            settings.termsAndConditionsData.ios.body.line3 undefined
          </div>
          <div
            style={
              Object {
                "color": "#DEDDDE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 35,
                "lineHeight": 49,
                "paddingBottom": 16,
                "textAlign": "left",
              }
            }
          >
            settings.termsAndConditionsData.ios.body.line4 undefined
          </div>
          <div
            style={
              Object {
                "color": "#DEDDDE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 35,
                "lineHeight": 49,
                "paddingBottom": 16,
                "textAlign": "left",
              }
            }
          >
            settings.termsAndConditionsData.ios.body.line5 undefined
          </div>
          <div
            style={
              Object {
                "color": "#DEDDDE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 35,
                "lineHeight": 49,
                "paddingBottom": 16,
                "textAlign": "left",
              }
            }
          >
            settings.termsAndConditionsData.ios.body.line6 undefined
          </div>
          <div
            style={
              Object {
                "color": "#DEDDDE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 35,
                "lineHeight": 49,
                "paddingBottom": 16,
                "textAlign": "left",
              }
            }
          >
            settings.termsAndConditionsData.ios.body.line7 undefined
          </div>
          <div
            style={
              Object {
                "color": "#DEDDDE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 35,
                "lineHeight": 49,
                "paddingBottom": 16,
                "textAlign": "left",
              }
            }
          >
            settings.termsAndConditionsData.ios.body.line8 undefined
          </div>
          <div
            style={
              Object {
                "color": "#DEDDDE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 35,
                "lineHeight": 49,
                "paddingBottom": 16,
                "textAlign": "left",
              }
            }
          >
            settings.termsAndConditionsData.ios.body.line9 undefined
          </div>
          <div
            style={
              Object {
                "color": "#DEDDDE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 35,
                "lineHeight": 49,
                "paddingBottom": 16,
                "textAlign": "left",
              }
            }
          >
            settings.termsAndConditionsData.ios.body.line10 undefined
          </div>
          <div
            style={
              Object {
                "color": "#DEDDDE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 35,
                "lineHeight": 49,
                "paddingBottom": 16,
                "textAlign": "left",
              }
            }
          >
            settings.termsAndConditionsData.ios.body.line11 undefined
          </div>
          <div
            style={
              Object {
                "color": "#DEDDDE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 35,
                "lineHeight": 49,
                "paddingBottom": 16,
                "textAlign": "left",
              }
            }
          >
            settings.termsAndConditionsData.ios.body.line12 undefined
          </div>
          <div
            style={
              Object {
                "color": "#DEDDDE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 35,
                "lineHeight": 49,
                "paddingBottom": 16,
                "textAlign": "left",
              }
            }
          >
            settings.termsAndConditionsData.ios.body.line13 undefined
          </div>
        </View>
      </View>
    </View>
  </RCTScrollView>
</div>
`;
